src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4363,6 +4398,9 @@ void os::init(void) {
 
   check_pax();
 
+  // Check the availability of MADV_POPULATE_WRITE.
+  FLAG_SET_DEFAULT(UseMadvPopulateWrite, (::madvise(0, 0, MADV_POPULATE_WRITE) == 0));
```

Comment:
Can we delay this to the first attempt? Switch it off if the first attempt returns EINVAL? Every system call saved at startup is good.

Reply:
Is that possible? Won't that clobber the user-supplied parameter?

Reply:
yes you are right. We should only try madvise if the switch is enabled. 

And we should do it here,  not as I suggested earlier on demand. Because changing flags after init is unexpected (eg PrintFlagsFinal).